sdk/tables/azure-data-tables/tests/test_retry.py

```diff
@@ -0,0 +1,188 @@
+        self.count += 1
+        response = super(RetryRequestTransport, self).send(request, **kwargs)
+        return response
+
+# --Test Class -----------------------------------------------------------------
+class StorageRetryTest(AzureTestCase, TableTestCase):
+
+    def _set_up(self, tables_storage_account_name, tables_primary_storage_account_key, url='table', default_table=True, **kwargs):
+        self.table_name = self.get_resource_name('uttable')
+        self.ts = TableServiceClient(
+            self.account_url(tables_storage_account_name, url),
+            credential=tables_primary_storage_account_key,
+            **kwargs
+        )
+        self.table = self.ts.get_table_client(self.table_name)
+        if self.is_live and default_table:
+            try:
+                self.ts.create_table(self.table_name)
+            except ResourceExistsError:
+                pass
+
+        self.query_tables = []
+
+    def _tear_down(self, **kwargs):
+        if self.is_live:
+            try:
+                self.ts.delete_table(self.table_name, **kwargs)
+            except:
+                pass
+
+            try:
+                for table_name in self.query_tables:
+                    try:
+                        self.ts.delete_table(table_name, **kwargs)
+                    except:
+                        pass
+            except AttributeError:
+                pass
+
+    # --Test Cases --------------------------------------------
+    @TablesPreparer()
+    def test_retry_on_server_error(self, tables_storage_account_name, tables_primary_storage_account_key):
+        self._set_up(tables_storage_account_name, tables_primary_storage_account_key, default_table=False)
+        try:
+            # Force the create call to 'timeout' with a 408
+            callback = ResponseCallback(status=201, new_status=500).override_status
+
+            new_table_name = self.get_resource_name('uttable')
+            # The initial create will return 201, but we overwrite it and retry.
+            # The retry will then get a 409 and return false.
+            with self.assertRaises(ResourceExistsError):
+                self.ts.create_table(new_table_name, raw_response_hook=callback)
+        finally:
+            self.ts.delete_table(new_table_name)
+            self._tear_down()
+
+
+    @TablesPreparer()
+    def test_retry_on_timeout(self, tables_storage_account_name, tables_primary_storage_account_key):
+        retry = ExponentialRetry(initial_backoff=1, increment_base=2)
+        self._set_up(tables_storage_account_name, tables_primary_storage_account_key, retry_policy=retry, default_table=False)
+
+        new_table_name = self.get_resource_name('uttable')
+        callback = ResponseCallback(status=201, new_status=408).override_status
+
+        try:
+            # The initial create will return 201, but we overwrite it and retry.
+            # The retry will then get a 409 and return false.
```

Comment:
the callback says `new_status=408` but your comment says the retry is a `409`, just for clarification is it expecing a 408 or 409?

Reply:
Actually both - I will update the comments to be clearer.